HULKs/hulk
# Improve twix parameter save diff

REQUIRES #248 
## Introduced Changes

Fullfills the following.

```rust
let stored_value = json!({"a":{"b":[1,2,3], "c":10}, "x":1000});
let incoming_sparse_value = json!({"a":{"b":[1,4,3], "c":10}});

// Only "b" has changes.
let expected_diff = json!({"a":{"b":[1,4,3]}});

assert_eq!(
    get_diff_against_stored_value(&stored_value, &incoming_sparse_value),
    expected_diff
);
```
See the issue for further details.


Fixes #240 

## ToDo / Known Issues

We still directly write to head json of a particular robot. Locations are not respected (unlike `serialize()` of parameters crate (Moved from communication server in #248 ))

## Ideas for Next Iterations (Not This PR)

* Further refactoring on these crates and how they are used could be done.

## How to Test

Cargo test on twix should do the trick ;)
OR
Open twix open manual calibration, change only extrinsic parameters.
Once you click save, only extrinsic parameter changes should show up in the json file. (Before this fix, both intrinsic and extrinsic showed up.)
